presentations/debugging-rust/slides.adoc

```diff
@@ -0,0 +1,189 @@
+
+== LLDB
+
+*   *Default choice for Rust.*
+*   Part of LLVM that Rust uses for compilation.
+*   Used to support many languages, but the team decided to focus on C, C++, and Objective C only.
+*   Has extension API for supporting other languages, which is not enough for Rust.
+
+== LLDB
+
+Rust project maintains a fork of LLDB with extended support for the language.
+
+*   Part of overall LLVM fork.
+*   Constantly updated and well-maintained.
+*   Non-Rust-specific bug fixes get upstreamed to main LLVM repository
+
+== Wrappers
+
+Rust comes with `rust-gdb` and `rust-lldb` wrappers around debuggers.
+
+They improve visualizing Rust values printed in console.
+
+== Editors and IDEs
+
+*Rust-analyzer* does not come with debugger support on its own.
+
+Instead it relies on other editor / ide plugins for debugging support.
+
+Prompts you to install one when you open a Rust project.
+
+== VSCode Extensions
+
+*   https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools[Microsoft C/C++ extension].
+*   https://marketplace.visualstudio.com/items?itemName=webfreak.debug[Native Debug].
+*   https://marketplace.visualstudio.com/items?itemName=vadimcn.vscode-lldb[CodeLLDB].
+
+== CodeLLDB.
+
+*   LLDB-only.
+*   *Maintains it's own fork of Rust's LLDB with even more Rust enhancements!*
+*   Downloads it on first installation.
+*   Seamless debugging experience.
+
+== !
+
+Both  Microsoft C/C++ and Native Debug support GDB _and_ LLDB.
+
+Microsoft's extension offers better support for displaying PDB information on Windows.
+
+== IntelliJ-Rust
+
+*   A plugin for IDEA and CLion
+*   Produced by JetBrain.
+*   Like CodeLLDB also maintains it's own fork of Rust's LLDB for better DX.
+*   Requires a JB license.
+
+== What to use?
+
+*   *VSCode + CodeLLDB offer the best debugging experience across all platforms.*
+*   https://docs.microsoft.com/en-us/windows/dev-environment/rust/setup[Microsoft recommends CodeLLDB even for Windows use.]
+*   IntelliJ-Rust is great if that's your IDE of choice.
+*   Native Debug and Microsoft C/C++ extensions can work for you on platforms where only GDB is available.
+
+== `rr`
+
+*   https://rr-project.org[A Linux-only terminal-based time-traveling debugger.] 
+*   Uses GDB under the hood, supports Rust.
+*   https://pernos.co[Pernosco] - GUI debugging tool on top of `rr` on top of `gdb` - offers Rust support, too.
+*   May help you in very difficult situations.
+
+== Things may not work well
+
+*   PDB may result in subpar debugging experience.
+**  If possible try debugging your code on OSes other than Windows 
+**  Or try using GNU-based toolchain on Windows.
+*   Watch expressions are limited. 
+**  Can't use `match` or `if` expressions
+**  Some method calls may not produce results.
+*   Some values can't be shown: function preferences, closures.
+*   Breakpoints may sometimes not work in closures and in async code.
+*   Trait objects and trait methods may be difficult for debugger to resolve.
+*   ...
```

Comment:
Instead of these `..` can we either add extra points to Speaker notes or just remove this. Other trainers may not  know what to do with these ellipses.